.github/actions/sync-to-drive-action/syncgdrive.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+import google.auth
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+import argparse
+
+"""
+Google Drive Folder ID = 16eBcMpkgMTV9mlKxYmNda64X_dmCFdkk
+"""
+
+def list_to_dict(list1):
+    """Return list of key-value pairs: key=filename, value=''"""
+    return [{l: ''} for l in list1]
+
+def fetch_drive_files():
+    """
+    Google Drive API Client
+
+    Return a list of key-value pairs from the response: key=filename, value=id
+    """
+    creds, _ = google.auth.default()
+
+    try:
+        # google drive api client
+        service = build('drive', 'v3', credentials=creds)
+        response = service.files().list(q="'16eBcMpkgMTV9mlKxYmNda64X_dmCFdkk' in parents").execute()
+        files = [{file.get('name'): file.get('id')} for file in response.get('files', [])]
+
+    except HttpError as error:
+        print(F'An error occurred: {error}')
+        files = None
+
+    return files
+
+def compare_dicts(dict1, dict2):
```

Comment:
Instead of writing your own implementation of diffing, you should switch to using sets and use standard python set operators.